File: Semestrul 3/MAP/lab 2 MAP/model/OldQueueContainer.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;

public class OldQueueContainer implements Container{
    List<Task> tasks = new ArrayList<>();

    @Override
    public Task remove() {
        if (!tasks.isEmpty()) {
            return tasks.remove(0);
        } else {
            return null;
        }
    }

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}

File: Semestrul 3/MAP/lab 10 MAP/model/OldQueueContainer.cs
using System.ComponentModel;

namespace seminar10.model;

public class OldQueueContainer : IContainer
{
    private readonly List<Task> _tasks = [];
    
    public Task Remove()
    {
        if (_tasks.Count != 0) return null!;
        var task = _tasks.ElementAt(0);
        _tasks.RemoveAt(0);
        return task;
    }

    public void Add(Task task)
    {
        _tasks.Add(task);
    }

    public int Size()
    {
        return _tasks.Count;
    }

    public bool IsEmpty()
    {
        return _tasks.Count == 0;
    }
}